app/controllers/api/base_controller/generic.rb

```diff
@@ -94,6 +94,23 @@ def delete_resource(type, id = nil, _data = nil)
         delete_resource_action(klass, type, id)
       end
 
+      def copy_resource(type, id, data = {})
```

Comment:
It looks pretty general, so I thought maybe it can be useful in the future if you want to add copy API for some entity.
What do you think?